runtime/src/accounts_db.rs

```diff
@@ -1115,9 +1156,21 @@ impl AccountsDB {
             }
             if !account_maps.is_empty() {
                 accounts_index.roots.insert(*slot_id);
-                let mut _reclaims: Vec<(u64, AccountInfo)> = vec![];
+                let mut reclaims: Vec<(u64, AccountInfo)> = vec![];
                 for (pubkey, (_, account_info)) in account_maps.iter() {
-                    accounts_index.insert(*slot_id, pubkey, account_info.clone(), &mut _reclaims);
+                    accounts_index.insert(*slot_id, pubkey, account_info.clone(), &mut reclaims);
+                }
+                for (slot_id, account_info) in reclaims {
+                    if let Some(slot_storage) = storage.0.get(&slot_id) {
+                        if let Some(store) = slot_storage.get(&account_info.id) {
+                            assert_eq!(
+                                slot_id, store.slot_id,
+                                "AccountDB::accounts_index corrupted. Storage should only point to one slot"
+                            );
+                            store.remove_account();
```

Comment:
This maintains the count of count_and_status of for updated non-zero-lamports accounts.